### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by FernandesMF**
This PR updates the MintMaker docs. There are new topics, and also some previous topics have been moved around. I made this reorganization because I felt the new topics did not quite fit in the old organization.

Here's a temptative list of changes:
- reorganization of the topics:
    - 'supported managers' now goes after 'configurations'
    - sections on offboarding and `updateNotScheduled`, previously under 'configuration', are now under 'advanced topics'
- moved the section on how to create custom configs to its own page and expanded a bit
- introduction is also expanded
- new section on how to point mintmaker to custom container/docker files
- new section about automerging under 'advanced topics'

I believe I haven't been consistent regarding the Renovate vs MintMaker nomenclature, so please be extra harsh about it! 😂 

Since there was a bit of moving around, I also suggest you `mdbook serve` the book as it is in this branch to have a better grasp of the new organization. I guess that is hard to check from the markdowns alone.

I'm also not sure about grouping several topics under 'advanced topics'. Perhaps they deserve their own page as well... let me know what you think!

Thanks! 😁

### CODE REVIEW

File: src/user/advanced_topics.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+The default value of `updateNotScheduled` is `true`, which leads to this behavior that might seem unexpected at first.
+
+## Automerge
+
+It is possible to configure Renovate to merge updates automatically for specific
+dependencies. You can find the documentation on this topic [here](https://docs.renovatebot.com/key-concepts/automerge/).
+
+When set in a given PR/MR, an automerge will happen provided two conditions are met:
+- the repositorie's CI pipeline ran successfully succeeds, and
```
Comment (staticf0x) [inline src/user/advanced_topics.md:48]:
```suggestion
- the repository CI pipeline ran successfully, and
```

File: src/user/advanced_topics.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+Because of the need for the CI pipeline to succeed, you should expect that the
+merge will only happen in the next time mintmaker is run.
+
+This following timeline exemplifies the events leading to an automerge in a repository:
+
+  Time  | Event
+--- | --- 
+__[10:00am]__ | MintMaker run 1 starts 
+__[10:01am]__ | PR for dependency `xyz` in is filed
```
Comment (staticf0x) [inline src/user/advanced_topics.md:59]:
```suggestion
__[10:01am]__ | PR for dependency `xyz` is filed
```

File: src/user/advanced_topics.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+__[10:02am]__ | CI pipeline is started
+__[10:05am]__ | CI pipeline finishes successfully
+__[10:10am]__ | MintMaker run 1 is finished
+... | ...
+__[11:00am]__ | MintMaker run 2 starts
+__[11:01am]__ | PR for dependency `xyz` is detected
+__[11:02am]__ | PR for dependency `xyz` is automerged
```
Comment (staticf0x) [inline src/user/advanced_topics.md:66]:
```suggestion
__[12:00pm]__ | MintMaker run 2 starts
__[12:01pm]__ | PR for dependency `xyz` is detected
__[12:02pm]__ | PR for dependency `xyz` is automerged
```

I would prefer to use the actual times in current MintMaker so we don't give users hope that it runs every hour :smile:

**Review by staticf0x — CHANGES_REQUESTED** [review]
Looks good, it makes sense to have it organized this way, thanks @FernandesMF. My comments are mostly just about nitpicks or typos. One thing I noticed though, did you intend on keeping the Supported managers page or not? It's still in the repo, but I remember there was a discussion about removing it, so just asking.

File: src/user/introduction.md [hunk 4]
```diff
@@ -2,3 +2,20 @@
...
+
+Renovate is a tool that automates dependency updates in your repository. Its goal
+is to make dependency management easier and a continued practice. That way, your
+project will aways have the latest fixes for bugs and security issues.
+
+You can find interesting use cases and motivations for using Renovate in your
+project [here](https://docs.renovatebot.com/user-stories/swissquote/).
+
+MintMaker is the integration of Renovate to Konflux CI, plus an extension to
+RPM lockfiles.
```
[the remark this answers is not in the document]
Comment (FernandesMF) [reply src/user/introduction.md]: Done!

File: src/user/custom_containerfiles.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
+# Custom Container Files
```
[the remark this answers is not in the document]
Comment (FernandesMF) [reply src/user/custom_containerfiles.md]: Done!

File: src/SUMMARY.md [hunk 6]
```diff
@@ -3,8 +3,10 @@
 # User
 
 - [Introduction](./user/introduction.md)
-- [Supported managers](./user/managers.md)
 - [Configuration](./user/configuration.md)
+- [Supported managers](./user/managers.md)
+- [Custom Container files](./user/custom_containerfiles.md)
```
[the remark this answers is not in the document]
Comment (FernandesMF) [reply src/SUMMARY.md]: Done!

File: src/user/advanced_topics.md (hunk 2 above)
Comment (FernandesMF) [reply src/user/advanced_topics.md]: Done!

File: src/user/advanced_topics.md (hunk 3 above)
Comment (FernandesMF) [reply src/user/advanced_topics.md]: Very good point indeed 😂

File: src/user/advanced_topics.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
+# Advanced Topics
```
[the remark this answers is not in the document]
Comment (FernandesMF) [reply src/user/advanced_topics.md]: Done!

**Comment by FernandesMF** [comment]
Thank you for the thorough review, @staticf0x! Good catches, and good point about the "fake schedule" 😂 

About the managers page, I was forgetting about it! I removed it from `summary.md`, but I left the file in the repo still. I believe it might be useful in case we want to rewrite that section, maybe it is an okay starting point.

Could you have another look? Thanks!

**Comment by FernandesMF** [comment]
You're absolutely right, @staticf0x! I'll remove that file and proceed with the merge. Thanks a lot! 😁

**Pull request merged**